libgadget/utils/endrun.c

```diff
@@ -189,9 +189,6 @@ messagev(int crash, int where, const char * fmt, va_list va)
         sprintf(prefix, "[ %09.2f ] ", MPI_Wtime() - _timestart);
     }
 
-    if(!crash && where <= 0)
-        MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
The main purpose is to assert that if we see this message, then all ranks have proceeded to this point. 

I think this is still a useful assertion to have, but I agree we may have abused it.

What about not having the barrier in message(0), but adding a MPI_Barrier() before any message(0) when it is clearly an assertion of code progress?

Reply:
That makes sense - I think the problem is that message(0) means two separate things: 1) Assert all ranks are here. 2) Output this information only on Rank 0.